File: workshop/session-1/python/main.py
#!/usr/bin/env python3
"""Simple command line script to ask questions to AISHE server."""

import sys


def main():
    """Ask a question to AISHE server and print the result."""
    # Check if question was provided
    if len(sys.argv) < 2:
        print("Usage: python main.py <your question>")
        print("Example: python main.py 'What is the capital of France?'")
        sys.exit(1)

    # Get question from command line arguments
    question = " ".join(sys.argv[1:])

    print(f"Asking: {question}")
    print("Waiting for response...\n")

    raise Exception("not implemented ...")


if __name__ == "__main__":
    main()

File: workshop/session-1/java/src/main/java/Main.java
/**
 * Simple command line application to ask questions to AISHE server.
 *
 * Usage: java Main <your question>
 * Example: java Main "What is the capital of France?"
 */
public class Main {
    public static void main(String[] args) {
        // Check if question was provided
        if (args.length == 0) {
            System.out.println("Usage: java Main <your question>");
            System.out.println("Example: java Main \"What is the capital of France?\"");
            System.exit(1);
        }

        // Get question from command line arguments
        String question = String.join(" ", args);

        System.out.println("Asking: " + question);
        System.out.println("Waiting for response...\n");

        throw new UnsupportedOperationException("not implemented ...");
    }
}
